scipy/optimize/_constraints.py

```diff
@@ -249,6 +250,29 @@ def __init__(self, constraint, x0, sparse_jacobian=None,
         self.bounds = (lb, ub)
         self.keep_feasible = keep_feasible
 
+    def violation(self, x):
+        """How much the constraint is exceeded by.
+
+        Parameters
+        ----------
+        x : array-like
+            Vector of independent variables
+
+        Returns
+        -------
+        excess : array-like
+            How much the constraint is exceeded by, for each of the
+            constraints specified by `PreparedConstraint.fun`.
+        """
+        with suppress_warnings() as sup:
+            sup.filter(UserWarning)
+            ev = self.fun.fun(np.asarray(x))
```

Comment:
Would it be harmful to pass subclasses through with `np.asanyarray`? Don't know if that ever happens in practice / is useful for `optimize` stuff, so maybe just confirm.

Reply:
The array-like is passed to a user defined function, so it would be up to that user defined function as to whether it was acceptable. Internally `differential_evolution` uses `np.float64`, so the array will be that dtype. However I think internally other minimizers can use other dtypes.

Reply:
I was just a little worried that we might clobber ndarray subclasses with the choice of `asarray` instead of `asanyarray`, with no strong reason to chose the former.

Anyway, the reviews here are positive and there's a +1 from expert reviewer so in it goes!